exp/views/responses.py

```diff
@@ -39,773 +50,626 @@
-                        "age_in_days": age_in_days
-                        if "exact" in optional_headers
-                        else "",
-                        "age_rounded": str(round_age(int(age_in_days)))
-                        if "rounded" in optional_headers
-                        else "",
-                        "gender": resp.child.gender
-                        if "gender" in optional_headers
-                        else "",
-                        "language_list": resp.child.language_list
-                        if "languages" in optional_headers
-                        else "",
-                        "condition_list": resp.child.condition_list
-                        if "conditions" in optional_headers
-                        else "",
-                        "age_at_birth": resp.child.age_at_birth
-                        if "gestage" in optional_headers
-                        else "",
-                        "additional_information": resp.child.additional_information
-                        if "addl" in optional_headers
-                        else "",
-                    },
-                    "consent": resp.current_consent_details,
-                }
-            )
-    return json_responses
-
-
-def get_response_headers_and_row_data(resp=None):
-    # Note: this uses the actual response object rather than a dict returned by
-    # values() because we use several properties (which cannot be retrieved by
-    # values()):
-    #
-    # "withdrawn"
-    # "most_recent_ruling",
-    # "most_recent_ruling_arbiter",
-    # "most_recent_ruling_date",
-    # "most_recent_ruling_comment",
-    # "child__language_list",
-    # "child__condition_list"
-    #
-    # Iterating over the responses to fetch these properties would defeat the point
-    # so we just use the object.
-
-    age_in_days = (resp.date_created.date() - resp.child.birthday).days if resp else ""
-
-    all_row_data = [
-        ("response_id", resp.id if resp else "", "Short ID for this response"),
-        (
-            "response_uuid",
-            str(resp.uuid) if resp else "",
-            "Unique identifier for response. Can be used to match data to video filenames.",
+class ResponseDataColumn(NamedTuple):
+    # id: Unique key to identify data. Used as CSV column header and any portion before __ is used to create a
+    # sub-dictionary for JSON data.
+    id: str
+    description: str  # Description for data dictionary
+    extractor: types.LambdaType  # Function to extract value from response instance or dict
+    optional: bool = False  # is a column the user checks a box to include?
+    name: str = ""  # used in template form for optional columns
+    include_by_default: bool = False  # whether to initially check checkbox for field
+    identifiable: bool = False  # used to determine filename signaling
+
+
+# Columns for response downloads. Extractor functions expect Response instance
+response_columns = [
```

Comment:
If it's top/module level, capitalize it `RESPONSE_COLUMNS`

Reply:
thanks, done!